app/lib/features/pins/pages/pins_list_page.dart

```diff
@@ -0,0 +1,188 @@
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:atlas_icons/atlas_icons.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:flutter_staggered_grid_view/flutter_staggered_grid_view.dart';
+import 'package:go_router/go_router.dart';
+
+class PinsListPage extends ConsumerStatefulWidget {
+  final String? spaceId;
+
+  const PinsListPage({super.key, this.spaceId});
+
+  @override
+  ConsumerState<PinsListPage> createState() => _AllPinsPageConsumerState();
+}
+
+class _AllPinsPageConsumerState extends ConsumerState<PinsListPage> {
+  final TextEditingController searchTextController = TextEditingController();
+  bool hasSearchTerm = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: _buildAppBar(),
+      body: _buildBody(),
+    );
+  }
+
+  AppBar _buildAppBar() {
+    return AppBar(
+      centerTitle: false,
+      title: Column(
+        mainAxisAlignment: MainAxisAlignment.start,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Text(L10n.of(context).pins),
+          _buildSpaceName(),
+        ],
+      ),
+      actions: [
+        AddButtonWithCanPermission(
+          canPermissionParam: widget.spaceId != null
+              ? (spaceId: widget.spaceId!, canString: 'CanPostPin')
+              : null,
+          onPressed: () => context.pushNamed(
+            Routes.actionAddPin.name,
+            queryParameters:
+                widget.spaceId != null ? {'spaceId': widget.spaceId} : {},
+          ),
+        ),
+      ],
+    );
+  }
+
+  Widget _buildSpaceName() {
+    if (widget.spaceId != null) {
+      String spaceName =
+          ref.watch(roomDisplayNameProvider(widget.spaceId!)).valueOrNull ?? '';
+      return Text(
+        '($spaceName)',
+        overflow: TextOverflow.ellipsis,
+        style: Theme.of(context).textTheme.labelLarge,
+      );
+    } else {
+      return const SizedBox.shrink();
+    }
+  }
+
+  Widget _buildBody() {
+    AsyncValue<List<ActerPin>> pinList;
+
+    final searchValue = ref.watch(searchValueProvider) ?? '';
+    hasSearchTerm = searchValue.isNotEmpty;
+
+    if (hasSearchTerm) {
+      pinList = ref.watch(
+        pinListSearchProvider(
+          (spaceId: widget.spaceId, searchText: searchValue),
+        ),
+      );
+    } else {
+      pinList = ref.watch(pinListProvider(widget.spaceId));
+    }
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        _buildSearchBar(),
+        Expanded(
+          child: pinList.when(
+            data: (pins) => _buildPinsList(pins),
+            error: (error, stack) =>
+                Center(child: Text(L10n.of(context).loadingFailed(error))),
+            loading: () => Center(child: Text(L10n.of(context).loading)),
```

Comment:
Skeleton pls?!?

Reply:
Yes, for sure.

Reply:
Done!

<img width="386" alt="Screenshot 2024-07-10 at 9 34 48 PM" src="https://github.com/acterglobal/a3/assets/51526480/3bc94f66-1ea8-43f9-aaae-5c450313b0d6">